pkg/syncer/util.go

```diff
@@ -422,9 +425,65 @@ func getVcHostAndVolumeManagerForVolumeID(ctx context.Context,
 			return "", nil, logger.LogNewErrorf(log,
 				"could not get volume manager for the vCenter: %q", vCenter)
 		}
+
+		log.Debugf("Identified VC %s for multi VC setup for volume %s", vCenter, volumeID)
 		return vCenter, volumeManager, nil
 	}
 
 	return "", nil, logger.LogNewErrorf(log,
 		"failed to get VC host and volume manager. VolumeInfoService is not initialized.")
 }
+
+// getVcHostAndVolumeManagerFromTopologySegments returns VC host and the corresponding
+// volume manager that can access the given volume on VC.
+// Based on the nodeAffinity rules defined on the PV, this function finds out
+// the right VC. If overlapping topology segments are found in nodeAffinity,
+// error is returned.
+func getVcHostAndVolumeManagerFromTopologySegments(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *metadataSyncInformer) (string, volumes.Manager, error) {
```

Comment:
Can we break it down in two functions?

1. getTopologySegmentsFromNodeAffinityRules, which takes PV as an argument
2. getVcHostAndVolumeManagerFromTopologySegments which takes topology segments as an argument.